src/client/pages/OfferNew/Introduction/Sidebar/StickyBottomSidebar.tsx

```diff
@@ -41,32 +41,22 @@ const Cta = styled(Button)`
   }
 `
 
-type BooleanSetter = (state: boolean) => void
-const avoidDisplayNoneGlitch = (
-  setVisible: BooleanSetter,
-  setDisplayNone: BooleanSetter,
-  isVisible: boolean,
-) => () => {
-  if (isVisible) {
-    setDisplayNone(false)
-    setTimeout(() => setVisible(true), 20)
-  } else {
-    setVisible(false)
-    setTimeout(() => {
-      setDisplayNone(true)
-    }, 500)
-  }
-}
-
 export const StickyBottomSidebar: React.FC<Hidable & {
   onCheckoutOpen: () => void
 }> = ({ isVisible, onCheckoutOpen }) => {
   const [reallyIsVisible, setReallyIsVisible] = React.useState(false)
   const [displayNone, setDisplayNone] = React.useState(false)
-  React.useEffect(
-    avoidDisplayNoneGlitch(setReallyIsVisible, setDisplayNone, isVisible),
-    [isVisible],
-  )
+  React.useEffect(() => {
+    if (isVisible) {
+      setDisplayNone(false)
+      setTimeout(() => setReallyIsVisible(true), 20)
+    } else {
+      setReallyIsVisible(false)
+      setTimeout(() => {
+        setDisplayNone(true)
+      }, 500)
+    }
+  }, [isVisible])
```

Comment:
I dislike removing named functions like this. Now i don't understand why we do this little dance, before i understood that this was some kind of hack to fix some css-specific issue probably.

Reply:
I'm not sure I understand what you mean - you don't understand what it does now that it doesn't have a name? (Because I didn't understand it with the name either)

Reply:
Yes exactly :) without reading the whole function you can still kind of make out that it has to do with some kind of styling issue right? The naming gives this functionality kind of an explanation/context/purpose and I think named functions should be preferred to for example comments - comments would give the same benefits ish but have a high risk of becoming stale or outdated once you start changing the functionality later (especially since theyre commonly not on the same line meaning they’re invisible in git diffs and PRs, for example. Or at least that’s my experience. Please challenge it if you want!

Reply:
I generally don't like comments in code either and that's not what I'm suggesting. Maybe a better solution generally is to declare _and_ call functions inside `useEffect` if that's the only place where they're used (like they recommend in the docs) instead of just doing stuff inside the `useEffect` (which also works and is clear enough in many cases IMO unless there are very complicated things going on).
That being said - this is not a good example of a self-explanatory function name since it's still such a strange component and it's impossible at least for me to understand it's purpose from just the name (or the implementation).
Also, this was the original linting warning and I'm not sure what a better solution would have been than doing as the linter suggests:
![image](https://user-images.githubusercontent.com/42962286/97328446-50753c00-1876-11eb-8af6-18988717acef.png)